src/main/java/soot/toolkits/exceptions/ThrowableSet.java

```diff
@@ -456,19 +457,20 @@ public ThrowableSet add(RefType e) throws ThrowableSet.AlreadyHasExclusionsExcep
       }
     }
     FastHierarchy hierarchy = Scene.v().getOrMakeFastHierarchy();
+    boolean eHasNoHierarchy = !(e.getSootClass().hasSuperclass() || //
+        SootClass.JAVA_LANG_OBJECT.equals(e.getSootClass().getName()));
```

Comment:
Using the 'Object' type from Scene, you could cache the `SootClass` which you can derived from it and then do an object comparison `e.getSootClass() == OBJECT_CLASS`. This should be much faster then doing this string comparison and since `SootClass`should be managed by the `Scene`, you are guaranteed to only have one instance of the `Object` class

Reply:
Btw. there is a `//` at the end of this line!

Reply:
You're welcome! 
I think it would make sense to cache the result of `Scene.v().getObjectType().getSootClass()` since the call still involves some unecessary overhead due to map lookup and sanity checks.
I guess it's debatable which version is more readable. Though, according to our styling conventions, operators should be placed after the line break in such cases: https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break